### PR TITLE
Sorting operators for flowable/observable with specified comparable key

### PR DESCRIPTION
**Pull request by devDebajo**
By default, RxJava does not allow to set a lambda, which specify key to sort the observables

These 4 simple operators will help greatly simplify lambdas-comparators for sorting observables

### CODE REVIEW

**Review by RivuChk — APPROVED** [review]

File: src/main/kotlin/io/reactivex/rxkotlin/flowable.kt [hunk 1]
```diff
@@ -187,3 +187,21 @@ fun <A: Any, B: Any> Flowable<Pair<A, B>>.toMultimap() = toMultimap({it.first},{
...
+inline fun <T : Any, K : Comparable<K>> Flowable<T>.sortedBy(crossinline keySelector: (T) -> K): Flowable<T> {
+    return sorted { left, right -> keySelector(left).compareTo(keySelector(right)) }
+}
+
+/**
+ * Sort flowable descending with specified comparable key
+ */
+@CheckReturnValue
+@SchedulerSupport(SchedulerSupport.NONE)
+inline fun <T : Any, K : Comparable<K>> Flowable<T>.sortedByDescending(crossinline keySelector: (T) -> K): Flowable<T> {
```
Comment (repitch) [inline src/main/kotlin/io/reactivex/rxkotlin/flowable.kt:220]: Maybe it'll be better to add extra param `ascending: Boolean = true` to sortedBy, to combine those two functions in one?
(.. and the same for observable.kt)

Comment (devDebajo) [reply src/main/kotlin/io/reactivex/rxkotlin/flowable.kt:220]: No

**Comment by isfaaghyth** [comment]
great idea! please fix conflicting files @nikitosbobin and please consideration to merge it @JakeWharton.

**Review by isfaaghyth — APPROVED** [review]

**Comment by vpriscan** [comment]
Hi @nikitosbobin , you should pull upstream 2.x branch again to apply the fix for Travis build, there was a problem with JDK version (oraclejdk8 license was changed and it stopped working).

**Comment by devDebajo** [comment]
> Hi @nikitosbobin , you should pull upstream 2.x branch again to apply the fix for Travis build, there was a problem with JDK version (oraclejdk8 license was changed and it stopped working).

Okay, I`ll try to do this

**Comment by vpriscan** [comment]
> > Hi @nikitosbobin , you should pull upstream 2.x branch again to apply the fix for Travis build, there was a problem with JDK version (oraclejdk8 license was changed and it stopped working).
> 
> Okay, I`ll try to do this

I actually forgot to add, this is only for the build to pass. If you want your PR to be accepted, you should update your PR to target the new RxKotlin 3.x version. Version 2.x is now in maintenance mode, meaning no new features will be accepted.

**Comment by devDebajo** [comment]
> > > Hi @nikitosbobin , you should pull upstream 2.x branch again to apply the fix for Travis build, there was a problem with JDK version (oraclejdk8 license was changed and it stopped working).
> > 
> > 
> > Okay, I`ll try to do this
> 
> I actually forgot to add, this is only for the build to pass. If you want your PR to be accepted, you should update your PR to target the new RxKotlin 3.x version. Version 2.x is now in maintenance mode, meaning no new features will be accepted.

Got it

**Pull request closed without merging**